### ISSUE TITLE
Don't close modal on Esc when modal is unnamed

### ISSUE THREAD

**Issue by robsonrosa** [meanie/angular-modal#2]
I'm trying to use the modal in the simplest possible way. The code that opens the modal is just like that:
```
var app = angular.module('app', ['Modal.Service']);
app.controller('MainCtrl', function($scope, $modal) {
  this.openModal = () => {
    $modal.open({
      templateUrl: 'modal.html'
    });
  }
});
```

Unfortunatelly, the closeOnEsc options doesn't work in this case. Here, a [plunker](https://plnkr.co/edit/BHzysXKv3NEH6TEkUrwj?p=preview).
I've checked the code and realized that the `closeOnEsc` options only works if the modal has a name. The `open` function in the `$modal` service expects three arguments: `name`, `options` and `closeOthers`. So I've tried this way:
```
var app = angular.module('app', ['Modal.Service']);
app.controller('MainCtrl', function($scope, $modal) {
  this.openModal = () => {
    $modal.open('my-modal', {
      templateUrl: 'modal.html'
    });
  }
});
```

In this case, an error occurs and no modal is shown. So, I did some hack in the `meanie-angular-modal` checking if a name is given, before cheking the modal stacking. Here, the code hacked:

Before:
```
//Close on escape?
if (options.closeOnEsc) {
  modal.closeOnEsc = function (event) {
    var key = event.keyCode || event.which;
    if (key === 27 && $modalStack.isLast(name)) {
      $rootScope.$apply(function () {
        closeModal(modalInstance, 'cancel', true);
      });
    }
  };
  $document[0].addEventListener('keydown', modal.closeOnEsc);
}
```

Now:
```
//Close on escape?
if (options.closeOnEsc) {
  modal.closeOnEsc = function (event) {
    var key = event.keyCode || event.which;
    if (key === 27 && (!name || $modalStack.isLast(name))) {
      $rootScope.$apply(function () {
        closeModal(modalInstance, 'cancel', true);
      });
    }
  };
  $document[0].addEventListener('keydown', modal.closeOnEsc);
}
```

It works like a charm.

**Comment by adamreisnz** [comment]
Cool, thanks for finding this. I'll have a look at it soon and release a fix.

I only use the module with named modals (and I have a base modal controller that can be extended in case you only want simple modals), and I find that a more effective way of using it.

If you wan't, I'm happy to share a code example of that with you.

**Comment by adamreisnz** [comment]
@robsonrosa I've used your fix and released this as 1.11.1, but be aware that a side effect will be that if you open multiple unnamed modals, pressing ESC will close them all.

Here's a code sample of a base modal controller that you could use (stripped a bit).

```js
/**
 * Module definition and dependencies
 */
angular.module('Shared.Modal.Controller', [])

/**
 * Config
 */
.config($modalProvider => {
  $modalProvider.modal('basic', {
    controller: 'ModalCtrl',
  });
})

/**
 * Controller
 */
.controller('ModalCtrl', function($modalInstance) {

  /**
   * On init
   */
  this.$onInit = function() {
    this.isConfirming = false;
  };

  /**
   * Confirm
   */
  this.confirm = function() {

    //Async handler?
    if (this.handler && angular.isFunction(this.handler)) {
      this.isConfirming = true;
      return this.handler(this.model)
        .then(result => $modalInstance.close(result))
        .finally(() => this.isConfirming = false);
    }

    //Sync
    $modalInstance.close(this.model);
  };

  /**
   * Close
   */
  this.close = this.cancel = function() {
    $modalInstance.dismiss('cancel');
  };
});
```

It defines a "basic" named modal which you can then use to open with various templates:

```js
$modal.open('basic', {
  templateUrl: 'some-template.html',
  locals: {
    handler() {
        return doSomething();
    },
  },
});
```

This way you keep it simple, and don't need to define multiple named modals if they're quite simple and don't need their own controller.

And if they do, you can simply extend the controller:

```js
/**
 * Custom modal controller
 */
.controller('SomeOtherModalCtrl', function($controller, $modalInstance) {

  //Get controllers
  const $ctrl = this;
  const $base = $controller('ModalCtrl', {$modalInstance});

  //Extend
  angular.extend($ctrl, $base);

  //...add other functionality
});
```

Hope this helps.

**Comment by robsonrosa** [comment]
Of course it helped. :)

Thanks a lot for the quick fix and for sharing your approach on handling modals.
